src/common/HTTPRequest.ts

```diff
@@ -295,12 +382,45 @@ export class HTTPRequest {
    * ```
    *
    * @param overrides - optional overrides to apply to the request.
+   * @param priority - If provided, intercept will be resolved using
+   * coopeative handling rules. Otherwise, intercept will be resovled
+   * immediately.
```

Comment:
```suggestion
   * @param priority - If provided, intercept is resolved using
   * cooperative handling rules. Otherwise, intercept is resolved
   * immediately.
```